packages/components/pie-cookie-banner/src/defs.ts

```diff
@@ -63,7 +63,13 @@ export interface CookieBannerProps {
      */
     cookieTechnologiesLink: string;
 
-    defaultPreferences: Record<PreferenceIds, boolean> | object
+    /**
+     * Allows consumers to pass in specific preference(s) to the component which will toggle
+     * the switch to be on by default (if set to `true`).
+     *
+     * e.g. { 'functional': true }
+     */
+    defaultPreferences: Record<PreferenceIds, boolean> | NonNullable<unknown>;
```

Comment:
Can you please test in Aperture if Typescript correctly picks up the valid strings we can pass in? I noticed when testing yesterday it seemed to let me put anything in there but I think it should be telling me I'm not allowed if I did something like `{ foo: 'bar' }`

Reply:
Also out of interest what is `NonNullable<unknown>;` doing?

Reply:
Just wondering if something like `Partial<Record<PreferenceIds, boolean>>;` might be cleaner/achieve the same thing. Let me know what you think!

Reply:
The nonNullable removes null and undefined. Ill try the other suggestion tho. For the point about {foo:bar} are you suggesting we validate it? Our validation helper wont work in this situation. I wanted to split this prop back out into 3 values that are passed as individual props instead of a single prop then that way we can validate it properly so either we do that or we leave this as it is and add individual validation for this.

Reply:
I do not think we should validate it. My understanding is these Typings do not prevent incorrect values being passed at run time, just help during development. 

When you are working with this component in an IDE it should prevent you from adding { foo: 'bar' } to this prop, which is what I was asking about :)

I don't _think_ nonnullable is needed here but I think something like `Partial<Record<PreferenceIds, boolean>>;` could be nice

Reply:
@xander-marjoram so the prop is optional. Not forced to use. They can also pass in partial values so instead of the full set 3 keys they could pass in 2 or 1 key instead.

Reply:
@kevinrodrigues Optional makes sense to me, as long as we document the _default_ defaults well!

I think it could get confusing when a partial object is given, as then there could be values, default values and default default values all at the same time.